src/Application.cpp

```diff
@@ -381,12 +381,62 @@ namespace weatherserver {
     in this function. Locations will be added to it.
     @param UA_NodeId parentNodeId - The parent node id of the object where the variables will be added in OPC UA.
     */
-    static void requestLocations(UA_Server* server, CountryData& country, UA_NodeId parentNodeId) {
+    static void requestLocations(UA_Server& server, CountryData& country, const UA_NodeId& parentNodeId) {
         try {
             // ReSharper disable once CppExpressionWithoutSideEffects
             webService->fetchAllLocations(country.getCode(), country.getLocationsNumber()).then([&](web::json::value response) {
                 country.setLocations(LocationData::parseJsonArray(response));
 
+                /*
+                Trying to check if original LocationsNumber actually matches with records in json file. Adjust if needed.
+                */
```

Comment:
May be introduce new function for this new piece of code? The function `requestLocations` is already long enough.

Reply:
Had to change a few things again.

Just realized that check for difference is actually just 1 line of code since we have that locations number parameter in CountryData class. And if that parameter matches the vector size after parsing - no validation of the model needed. It takes that parameter as well when adding node to the model.

But if we have discrepancy either because of missing entries 207/193 or because of duplicates 193/188 we can go check the model number and correct it.

At the same time if we correct the model - need to correct the class parameter as well - added setter method (originally done through constructor).